scripts/gen_dynamic_openapi.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+set -e
+
+# Generate an OpenAPI document for all backends.
+#
+# Assumptions:
+#
+#   1. Vault has been checked out at an appropriate version and built
+#   2. vault executable is in your path
+#   3. Vault isn't already running
+#   4. Jq is installed
+
+echo "Starting Vault..."
+if pgrep -x "vault" > /dev/null
+then
+    echo "Vault is already running. Aborting."
+    exit 1
+fi
+
+vault server -dev -dev-root-token-id=root &
+sleep 2
+VAULT_PID=$!
+
+echo "Mounting all builtin backends..."
+
+# Read auth backends
+codeLineStarted=false
+regex='".*"'
+while read line; do
+    if [[ $line == *"credentialBackends:"* ]] ; then
+        codeLineStarted=true
+    elif [ $codeLineStarted = true ] && [[ $line = *"}"* ]]  ; then
+        break
+    elif [ $codeLineStarted = true ] && [[ $line =~ $regex ]] && [[ $line != *"Deprecated"* ]] ; then
```

Comment:
Maybe inline the regex since `Deprecated` is already inlined: `$line =~ '".*"'` (not sure if this works with shell). Otherwise, maybe call it `inQuotesRegex` or something similar to give an idea of what this is doing

Reply:
The inline gives me a shell check warning and doesn't seem to be matching a regex. I can rename it though.